gulpfile.js

```diff
@@ -1,122 +1,207 @@
+*/
+function buildCss (isDev) {
+	return obt.build.sass(gulp, obtBuildConfig(isDev))
+		.on('end', () => console.log('build-css completed'))
+		.on('error', err => console.warn('build-css errored', err));
+}
 
-const verifyCss = () => obt.verify.scssLint(gulp, {
-	sass: mainScssFilePath
-})
-.on('error', (e) => console.log('CSS linting error', e))
+/*
+ * A helper function which builds JS for development and production
+*/
+function buildJs (isDev) {
+	return obt.build.js(gulp, obtBuildConfig(isDev))
+		.on('end', () => console.log('build-js completed'))
+		.on('error', err => console.warn('build-js errored', err));
+}
 
-const verifyClientJs = () => obt.verify.esLint(gulp, {
-	js: mainJsFilePath,
-	excludeFiles: ['!public/**/**/*', '!src/js/svgloader.js']
-})
-.on('error', (e) => console.log('Client JS linting error', e))
+gulp.task('build:css:dev', () => buildCss(true));
 
-const verifyServerJs = () => obt.verify.esLint(gulp, {
-	js: serverJsPaths,
-	esLintPath: path.join(__dirname, '.eslintrc')
-})
-.on('error', (e) => console.log('Server JS linting error', e))
+gulp.task('build:js:dev', () => buildJs(true));
 
-gulp.task('build-css-dev', buildCss.bind(null, true));
+gulp.task('build:css:prod', () => buildCss(false));
 
-gulp.task('build-js-dev', buildJs.bind(null, true));
+gulp.task('build:js:prod', () => buildJs(false));
 
-gulp.task('build-css-prod', buildCss.bind(null, false));
+gulp.task('build', ['build:css:dev', 'build:js:dev']);
 
-gulp.task('build-js-prod', buildJs.bind(null, false));
+gulp.task('build:prod', ['build:css:prod', 'build:js:prod']);
 
-gulp.task('build', ['build-css-dev', 'build-js-dev']);
+gulp.task('default', ['watch']);
 
-gulp.task('build-prod', ['build-css-prod', 'build-js-prod']);
+gulp.task('serve', function () {
+	return nodemon({
+		script: './bin/www',
+		ext: 'js',
+		env: { 'NODE_ENV': 'development' }
+	});
+});
+
+gulp.task('start', ['watch']);
 
-gulp.task('nodemon', server);
+gulp.task('test:client', function (done) {
+	return new Server({
+		configFile: path.join(__dirname, 'karma.conf.js'),
+		singleRun: true
+	}, done)
+	.start();
+});
 
-gulp.task('tdd-client', done => {
-  new Server({
-    configFile: path.join(__dirname, 'karma.conf.js'),
-		singleRun: false
-  }, done).start();
+gulp.task('test:e2e', function () {
+	/*
+	 * Spins up the server in a child process ready for the tests to run against.
+	*/
+	function serve () {
+		const bin = path.join(__dirname, 'bin', 'www');
+		return spawn(bin, { detached: true})
+			.on('close', () => console.log("serve exited"))
+			.on('error', err => console.error(`"serve" Error: ${err}`));
+	}
+
+	const child = serve();
+
+	return installAndStartSelenium()
+		.then(function () {
+			return new Promise(function(resolve, reject) {
+				const wdioBin = path.join(__dirname, 'node_modules', '.bin','wdio');
+
+				/*
+				 * Executes WebDriverIO client binary to run our tests using the local Selenium server
+				*/
+				const wdio = spawn(wdioBin, ['wdio.conf.js'])
```

Comment:
There is a Gulp WDIO plugin but it does not inform you when WDIO has finished running the test suite. We need to know when WDIO has finished running the test suite so that we can shutdown the local Big FT server.